dataclients/kubernetes/kubernetestest/api.go

```diff
@@ -220,35 +219,56 @@ func parseSelectors(r *http.Request) map[string]string {
 	return selectors
 }
 
-// Filters all resources that are already set in k8s namespace using the given selectors map.
-// All resources are initially set to `namespace` as slices of bytes and for most tests it's not needed to make it any more complex.
-// This helper function deserializes resources, finds a metadata with labels in them and check if they have all
-// requested labels. If they do, they are returned.
-func filterBySelectors(resources []byte, selectors map[string]string) []byte {
-	if len(selectors) == 0 {
-		return resources
+func serve(w http.ResponseWriter, r *http.Request, resources []byte, name string) {
+	selectors := parseSelectors(r)
+	if name == "" && len(selectors) == 0 {
+		w.Write(resources)
+		return
 	}
 
-	labels := struct {
+	itemsMetadata := struct {
 		Items []struct {
 			Metadata struct {
+				Name   string            `json:"name"`
```

Comment:
Should we consider namespace as well?

Reply:
We can add it but is not used as the test api only filters by name or labels